src/tests/test_circumcision.py

```diff
@@ -0,0 +1,435 @@
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+    pop.circumcision.init_birth_circumcision_all(pop.data, pop.date)
+
+    # get stats
+    no_circumcised = len(pop.data[pop.data[col.CIRCUMCISED] == 1])
+    no_male = len(pop.data[pop.data[col.SEX] == SexType.Male])
+    mean = no_male * pop.circumcision.prob_birth_circ
+    stdev = sqrt(mean * (1 - pop.circumcision.prob_birth_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_circumcised, pop.data)
+    # check that no circumcised people have undergone VMMC
+    assert len(pop.data[pop.data[col.CIRCUMCISED] & pop.data[col.VMMC]]) == 0
+
+
+def test_birth_circumcision_stages():
+
+    N = 100000
+    start_date = date(2000, 1, 1)
+    stop_date = date(2020, 1, 1)
+    time_step = timedelta(days=90)
+
+    # build population
+    pop = Population(size=N, start_date=start_date)
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+    pop.circumcision.init_birth_circumcision_born(pop.data, pop.date)
+
+    # get stats
+    no_circumcised = len(pop.data[pop.data[col.CIRCUMCISED]])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] > 0.25)]
+    mean = len(male_population) * pop.circumcision.prob_birth_circ
+    stdev = sqrt(mean * (1 - pop.circumcision.prob_birth_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_circumcised, pop.data)
+
+    # evolve population
+    while pop.date <= stop_date:
+        # advance ages and birth circumcision
+        pop.data.age += time_step.days / 365
+        pop.circumcision.update_birth_circumcision(pop.data, time_step, pop.date)
+        pop.date += time_step
+
+    # get stats
+    no_circumcised = len(pop.data[pop.data[col.CIRCUMCISED]])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] > 0.25)]
+    mean = len(male_population) * pop.circumcision.prob_birth_circ
+    stdev = sqrt(mean * (1 - pop.circumcision.prob_birth_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_circumcised, pop.data)
+    # check that no circumcised people have undergone VMMC
+    assert len(pop.data[pop.data[col.CIRCUMCISED] & pop.data[col.VMMC]]) == 0
+
+
+def general_circumcision_checks(mean, stdev, no_circumcised, data):
+
+    # no female is marked as circumcised
+    assert len(data[(data[col.SEX] == SexType.Female)
+                    & data[col.CIRCUMCISED]]) == 0
+
+    # no uncircumcised people have circumcision dates
+    assert len(data[~data[col.CIRCUMCISED]
+                    & data[col.CIRCUMCISION_DATE].notnull()]) == 0
+
+    # all circumcised people have circumcision dates
+    assert len(data[data[col.CIRCUMCISED]]) == \
+           len(data[data[col.CIRCUMCISED] & data[col.CIRCUMCISION_DATE].notnull()])
```

Comment:
```suggestion
    assert (data[col.CIRCUMCISED] == data[col.CIRCUMCISION_DATE].notnull()).all()
```
Current test checks that circumcised => circumcision date but doesn't check that some people might have a circumcision date but not be circumcised. It is I think simpler to check that the boolean array for circumcision is equivalent to circumcision date being not null, and this will check for the implication in both directions.